HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+- Upon receiving this `AcceptUnsolicitedToken` HAPI command, the network will actually move the token from Airdrop-Sender's account to the Airdrop-Receiver's account and also add a token-association entry for this token on the receiver's account automatically. From this point onwards, the token behaves exactly as if the Airdrop-Recipient had manually created a token association and then received the token to the account.
+- The Airdrop-Recipient is responsible for renewal of the token association as part of their normal renewal after this point. Please note that the tokens are in the Airdrop-Sender's control until the Airdrop-Recipient calls the `AcceptUnsolicitedToken` API. As a result, they can also delete or burn these tokens while they are in the outbox. Doing so will remove the entries from the Unsolicited-inbox and Unsolicited-outbox automatically. Please note that these are conceptual names and the actual APIs/protobufs and REST APIs (mirror node) will be designed as this HIP develops further.
+
+### New APIs/UIs:
+
+- Services:
+    - `AcceptUnsolicitedToken` API
+- Mirror node:
+    - GET `unsolicitedInbox` query for the Airdrop-Recipient to get content of their unsolicited-inbox.
+    - GET `unsolicitedOutbox` query for the Airdrop-Sender to get content of their unsolicited-outbox.
+- Wallets and Explorers
+    - Wallets and Explorers should provide a way for users to be able to see the unsolicited-inbox and accept the tokens that they wish to accept
+
+### Other considerations
+
+- Auto-association: if the recipient of the token has *open* auto-association slots, then none of the mechanisms described in this document apply. That is treated as a regular transfer. The airdropped token will be auto-associated with the recipient and the transfer will succeed as it does today
+- Already associated token type: if the airdropped token type is already associated with the recipient’s account, then none of the mechanisms described in this document apply. That is treated as a regular transfer. The airdropped token will simply be transferred to the recipient's account as it does today.
+- Receiver-signature-required: if the Airdrop-Recipient has receiver-signature-required set on their account, then the airdrop will only succeed if they also sign the `AcceptUnsolicitedToken` API with their receiver-signature.
+- Custom fees on the token: The Airdrop-Sender will pay for the custom-fees levied on the token transfer at the time of the transfer (i.e. the time at which the Airdrop-Receiver signed the `AcceptUnsolicitedToken` transaction). Note that the custom-fees may change from the time the Airdrop-sender attempted to send the tokens to the time the actual transfer takes place, and it is a risk that the Airdrop-Sender will have to assume.
```

Comment:
@mgarbs @Ashe-Oro @lbaird 
How would this work with fallback fees? 

The main issue we had with NFT sending before wallets was that fallback fees require signature from both the sending account and the receiving account. This is because the receiver is required to pay the fallback fee in HIP-18. 

This is how Hashpack's Secure Trade broke the log jam of trading NFTs on Hedera again, as it gave an ability to gather both signatures to transfer NFTs with fallback fees. 

If I were to try to send a NFT with a fallback fee, does the Inbox change who is the payer of the fallback fee in that transaction? Or does this restrict all FTs and NFTs that have a fallback fee defined to not use the inbox?

Apologies on the tags are the right people, not sure who exactly is best from Swirlds to ask

Reply:
Three paths I think there are:

1. Allow users to accept transaction including fallback fees required from token that is in inbox 
2. Update HIP-18 to change account charged to be the sender account
3. Disallow tokens with fallback fees to use the inbox feature.

I like path 1 as it'll bring better UX to al use cases including trying to send a NFT with fallback fees for wallets on the network. 

Brings asynchronous signing that today is quite clunky with scheduled transactions on token transfer

Reply:
> Note that the custom-fees may change from the time the Airdrop-sender attempted to send the tokens to the time the actual transfer takes place, and it is a risk that the Airdrop-Sender will have to assume.

We may be able to improve this situation by layering in https://github.com/hashgraph/hedera-improvement-proposal/discussions/456

Reply:
@HGraphPunks inline with the inbox concept the tokens have not yet been sent. So fallback fees won't be applicable till the recipient accepts the token. At accept time it would behave exactly like a normal transfer.
As part of acceptance a user could be expected to review the token being accepted and would then be aware of the fallback fee they would have to pay.
Of course wallets and DApps would be encouraged to showcases the details to help users not miss this information.